torch/_decomp/decompositions.py

```diff
@@ -4343,6 +4343,55 @@ def _weight_norm_interface(x, y, dim):
     norm = x.norm(2, keep_dim, keepdim=True)
     return x * (y / norm), norm
 
+@register_decomposition(aten.isin)
+@out_wrapper()
+def isin(elements, test_elements, *, assume_unique=False, invert=False):
+    if test_elements.numel() < 10.0*torch.pow(torch.tensor(elements.numel(), dtype=torch.float32), torch.tensor(0.145)):
+        return isin_default(elements, test_elements, invert=invert)
+    else:
+        return isin_sorting(elements, test_elements, assume_unique=assume_unique, invert=invert)
+
+def isin_default(elements, test_elements, *, invert=False):
+    x = elements.view(*elements.shape, *((1,)*test_elements.ndim))
+    if not invert:
+        cmp = x == test_elements
+    else:
+        cmp = x != test_elements
+    dim = tuple(range(-1, -test_elements.ndim-1, -1))
+    return cmp.any(dim=dim)
+
+def isin_sorting(elements, test_elements, *, assume_unique=False, invert=False):
+    if assume_unique:
+        elements_flat = elements.flatten()
+        test_elements_flat = test_elements.flatten()
+    else:
+        elements_flat, unique_order = torch.unique(elements,
```

Comment:
Ah, apparently this cannot work because of the dynamic shapes. Should I just use the default algorithm when assume_unique=False?

Reply:
I think we could use this algorithm which doesn't need static shapes:
```python
sorted_test_elements = torch.sort(test_elements_flat)
idx = torch.searchsorted(sorted_test_elements, elements_flat)
test_idx = torch.where(idx < sorted_test_elements.numel() , idx, 0)
cmp = sorted_test_elements[test_idx] == elements_flat
cmp = cmp.logical_not() if invert else cmp
```

It's still O(n log n) but we should check the performance as well.

Reply:
I've added this, keeping the other version for `assume_unique=True`. We'll probably want some extensive test cases to benchmark this against. Let me know if something like that already exists.

Reply:
I think you will need to come up with your own test cases unfortunately, we don't have a very extensive benchmark suite. Mainly just benchmarks for key ML operations and models.